lib/cartopy/tests/test_img_nest.py

```diff
@@ -41,35 +42,49 @@
                               'wmts', 'aerial')
 
 
-def test_world_files():
-    func = cimg_nest.Img.world_files
-    fname = 'one'
-    expected = ['one.w', 'one.W', 'ONE.w', 'ONE.W']
-    assert func(fname) == expected
-
-    fname = 'one.png'
-    expected = ['one.pngw', 'one.pgw', 'one.PNGW', 'one.PGW',
-                'ONE.pngw', 'ONE.pgw', 'ONE.PNGW', 'ONE.PGW']
-    assert func(fname) == expected
-
-    fname = '/one.png'
-    expected = ['/one.pngw', '/one.pgw', '/one.PNGW', '/one.PGW',
-                '/ONE.pngw', '/ONE.pgw', '/ONE.PNGW', '/ONE.PGW']
-    assert func(fname) == expected
-
-    fname = '/one/two.png'
-    expected = ['/one/two.pngw', '/one/two.pgw',
-                '/one/two.PNGW', '/one/two.PGW',
-                '/one/TWO.pngw', '/one/TWO.pgw',
-                '/one/TWO.PNGW', '/one/TWO.PGW']
-    assert func(fname) == expected
-
-    fname = '/one/two/THREE.png'
-    expected = ['/one/two/THREE.pngw', '/one/two/THREE.pgw',
-                '/one/two/THREE.PNGW', '/one/two/THREE.PGW',
-                '/one/two/three.pngw', '/one/two/three.pgw',
-                '/one/two/three.PNGW', '/one/two/three.PGW']
-    assert func(fname) == expected
+@pytest.mark.parametrize('fname, expected', [
+    pytest.param('one', ['one.w', 'one.W', 'ONE.w', 'ONE.W']),
+    pytest.param('one.png', [
+        'one.pngw', 'one.pgw', 'one.PNGW', 'one.PGW', 'ONE.pngw', 'ONE.pgw',
+        'ONE.PNGW', 'ONE.PGW',
+    ]),
+    pytest.param('/one.png', [
+        '/one.pngw', '/one.pgw', '/one.PNGW', '/one.PGW', '/ONE.pngw',
```

Comment:
How about mutating the expected in the test based on Windows or not? Something like:

```
if sys.platform == 'win32':
    expected = ['C:' + pth.replace('/', '\') for pth in expected]
```

It would make things an awful lot simpler to see what is going on I think.

Reply:
Maybe, but weren't you also thinking of removing the case-changes on Windows?

Reply:
Yeah. It is an optimisation that *could* be made, but honestly I don't think it is worth the effort (even though it is only a single line change!).